packages/native/src/useLinking.tsx

```diff
@@ -592,7 +592,8 @@ export default function useLinking(
               // We couldn't find an existing entry to go back to, so we'll go back by the delta
               // This won't be correct if multiple routes were pushed in one go before
               // Usually this shouldn't happen and this is a fallback for that
-              await history.go(historyDelta);
+              // Add Math.max to guard going to < 0 index
+              await history.go(Math.max(historyDelta, 0));
```

Comment:
This is equivalent to `await history.go(0)` because `historyDelta < 0` in this part of the code

Reply:
@aldo-expensify Ahhh yes actually the histroy length is calculated in `go()` like 
```
if(n === 0) {
return
}
```
and a `if n<0 ` case is handeled here https://github.com/react-navigation/react-navigation/pull/10558#:~:text=Also%20the%20index%20%3D%20n%20%3C%200%20%3F%20Math.max(index%20%2D%20n%2C%200)%20%3A%20Math.min(index%20%2B%20n%2C%20items.length%20%2D%201)%3B which make sure that the length of history doesn't go below 0.

Reply:
> so index should be used in `window.history.go(index)`

the `go` method API accepts a delta to go forward or backward. it doesn't accept an index